docs/articles/user_acquisition_analytics.md

```diff
@@ -0,0 +1,448 @@
+    sources=[source_user], indices=[user_index], context_data=EXECUTOR_DATA
+)
+app: InMemoryApp = executor.run()
+```
+
+...and then input our user data.
+
+```python
+source_user.put([user_df])
+```
+
+ (The step above make take a few minutes or more. In the meantime, why not learn more about vectors in [Vectorhub](https://superlinked.com/vectorhub/).)
+
+## Load features
+
+Next, we collect all our vectors from the app.
+
+```python
+vs = VectorSampler(app=app)
+vector_collection = vs.get_all_vectors(user_index, user)
+vectors = vector_collection.vectors
+vector_df = pd.DataFrame(vectors, index=[int(id_) for id_ in vector_collection.id_list])
+vector_df.head()
+```
+
+```python
+vector_df.shape
+```
+
+Here are the first five rows (of 8000), and 776 columns, of the resulting dataframe:
+
+![](../assets/use_cases/user_acquisition_analytics/collected_vectors.png)
+
+## Clustering
+
+Next, we fit a clustering model.
+
+```python
+hdbscan = HDBSCAN(min_cluster_size=500, metric="cosine")
+hdbscan.fit(vector_df.values)
+```
+
+Let's create a DataFrame to store the cluster labels assigned by HBDSCAN and count how many users belong to each cluster:
+
+```python
+label_df = pd.DataFrame(
+    hdbscan.labels_, index=vector_df.index, columns=["cluster_label"]
+)
+label_df["cluster_label"].value_counts()
+```
+
+![user distribution by cluster label](../assets/use_cases/user_acquisition_analytics/user_distrib-by-cluster_label.png)
+
+## Visualizing the data
+
+Let's try to interpret our dataset by visualizing it using UMAP.
+
+First, we'll fit the UMAP and transform our dataset:
+
+```python
+umap_transform = umap.UMAP(random_state=0, transform_seed=0, n_jobs=1, metric="cosine")
+umap_transform = umap_transform.fit(vector_df)
+umap_vectors = umap_transform.transform(vector_df)
+umap_df = pd.DataFrame(
+    umap_vectors, columns=["dimension_1", "dimension_2"], index=vector_df.index
+)
+umap_df = umap_df.join(label_df)
+```
+
+Next, we join our dataframes and create a chart, letting us visualize the UMAP-transformed vectors, and coloring them with cluster labels.
+
+```python
+umap_df = umap_df.join(label_df)
+alt.Chart(umap_df).mark_circle(size=8).encode(
+    x="dimension_1", y="dimension_2", color="cluster_label:N"
+).properties(
+    width=600, height=500, title="UMAP Transformed vectors coloured by cluster labels"
+).configure_title(
+    fontSize=16,
+    anchor="middle",
+).configure_legend(
+    strokeColor="black",
+    fillColor="#EEEEEE",
+    padding=10,
+    cornerRadius=10,
+    labelFontSize=14,
+    titleFontSize=14,
+).configure_axis(
+    titleFontSize=14, labelFontSize=12
+)
+```
+
+![cluster visualization](../assets/use_cases/user_acquisition_analytics/cluster_visualization.png)
+
+The dark blue clusters (label -1) are outliers - not large or dense enough to form a distinct group. The large number of blobs indicates that 2/3 of the vector normal mass is comprised of *many* ad_creatives, while 1/3 is made up by just a *few*.
```

Comment:
I meant here that the vectors have 1/3 mass for each attribute, and for ad_creatives there are a limited number of distinct creatives, while signup_dates and activities are more continous - hence there are more dense blobs in the umap visualization